### ISSUE TITLE
Introspection of intermediate paths is not currently possible

### ISSUE THREAD

**Issue by colonelpanic8** [rblaze/haskell-dbus#13]
If you expose an object at /a/b/c

and request the path a/b
```
dbus-send --session           \
  --dest=interfacename\
  --type=method_call          \
  --print-reply               \
  /a/b       \
  org.freedesktop.DBus.Introspectable.Introspect
```

you get an unknown object response with any object exposed by this library.

The path concept is supposed to be hierarchical, but we represent things as a simple ObjectPath -> InterfaceMap. Well need to change the way that object paths are represented to get this to work.

I also think that it might be a good time to consider that interfaces and object paths ought to be more independent than they are in the current implementation. The reason they are separate concepts is that it is supposed to be possible to expose the same interface on different object paths. Its kinda like the class/object distinction.

**Comment by rblaze** [comment]
If you can send patch to support this, I'll be happy to release. My knowledge of dbus workings is rather limited, so I'm totally open to any improvements.

**Comment by colonelpanic8** [comment]
This was fixed by #12
